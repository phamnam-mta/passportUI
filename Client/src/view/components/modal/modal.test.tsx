import * as React from "react";
import { shallow } from "enzyme";

import Modal from "./modal";

describe("<Modal />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            header: <div>Title</div>,
            body: <div>Body Content</div>,
            footer: <div>Footer Message</div>,
            buttonGroup: (
                <React.Fragment>
                    <div>Button 1</div>
                    <div>Button 2</div>
                </React.Fragment>
            ),
            isOpen: true,
        };
    });

    describe("Rendering", () => {
        it("should match snapshot, when it does not have close button", () => {
            const wrapper = shallow(<Modal {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it has close button", () => {
            const props = {
                ...baseProps,
                onClose: () => {},
            };

            const wrapper = shallow(<Modal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it is closed", () => {
            const props = {
                ...baseProps,
                isOpen: false,
            };

            const wrapper = shallow(<Modal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it is loading", () => {
            const props = {
                ...baseProps,
                isLoading: true,
            };

            const wrapper = shallow(<Modal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it is draggable", () => {
            const props = {
                ...baseProps,
                isDraggable: true,
            };

            const wrapper = shallow(<Modal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when width is set", () => {
            const props = {
                ...baseProps,
                width: 700,
            };

            const wrapper = shallow(<Modal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
