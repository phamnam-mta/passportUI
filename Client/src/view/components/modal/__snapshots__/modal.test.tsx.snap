// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Rendering should match snapshot, when it does not have close button 1`] = `
<Modal
  isOpen={true}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 560,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> Rendering should match snapshot, when it has close button 1`] = `
<Modal
  isOpen={true}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 560,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
      <CloseButton
        onClick={[Function]}
      />
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> Rendering should match snapshot, when it is closed 1`] = `
<Modal
  isOpen={false}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 560,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> Rendering should match snapshot, when it is draggable 1`] = `
<Modal
  dragOptions={
    Object {
      "closeMenuItemText": "Close",
      "menu": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "moveMenuItemText": "Move",
    }
  }
  isOpen={true}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 560,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> Rendering should match snapshot, when it is loading 1`] = `
<Modal
  isOpen={true}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 560,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
    <div
      className="modal-loading"
    >
      <StyledSpinnerBase
        className="modal-spinner"
        size={2}
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> Rendering should match snapshot, when width is set 1`] = `
<Modal
  isOpen={true}
  styles={
    Object {
      "main": Object {
        "minHeight": "auto",
        "width": 700,
      },
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Body Content
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div>
        Footer Message
      </div>
      <div
        className="modal-button-group"
      >
        <div>
          Button 1
        </div>
        <div>
          Button 2
        </div>
      </div>
    </div>
  </div>
</Modal>
`;
